raritan/decorators.py

```diff
@@ -154,7 +155,22 @@ def wrapper_function(*args, **kwargs):
             # Assets are listed in two tiers.
             for group, assets in sources.items():
                 for key, name in assets.items():
+                    if isinstance(name, dict):
+                        # If assets is a dictionary, iterate over its items
+                        inner_optional_flag = name.get('optional', False)
+                        name = name.get('file')
+                    else:
+                        inner_optional_flag = optional_flag  # Use the default optional_flag
                     logger.info(f'Handling asset: {name}')
+                    # Check the optional flag
+                    if not os.path.exists(group + '/' + name):
+                        # It is not optional
+                        if not inner_optional_flag:
+                            error(f"Non-Optional file missing: {name}")
```

Comment:
Will this cause the flow to fail? If a non-optional file is missing, we want the whole thing to crash.

Reply:
Yes this had just printed out to console that an non-optional file was missing and would continue on. I just added a FileNotFound error, which will cause the flow to fail.

Good catch!